zcash_primitives/src/zip32.rs

```diff
@@ -233,6 +235,41 @@ pub fn sapling_default_address(
     sapling_find_address(fvk, dk, DiversifierIndex::new()).unwrap()
 }
 
+/// Returns the internal full viewing key and diversifier key
+/// for the provided external fvk and dk
+pub fn sapling_derive_internal_fvk(
```

Comment:
Document what this API is for (specifically implementing UFVKs, since otherwise this would be duplicative of `ExtendedFullViewingKey::derive_internal`).

Reply:
We should add the `DiversifiableFullViewingKey` type here as we have in zcashd, and then this should be a member function of that type. If we had that, then the `ExtendedFullViewingKey` impl could be removed.

Reply:
Also, it's not critical for the moment but now that zip32 has non-Sapling stuff in it, I'd really like this to be moved into the Sapling module.